depmapomics/copynumbers.py

```diff
@@ -0,0 +1,479 @@
+#cn.py
+
+import numpy as np
+from gsheets import Sheets
+from depmapomics import tracker as track
+from depmapomics import utils
+from depmapomics.config import *
+from IPython.display import Image, display
+import dalmatian as dm
+import pandas as pd
+from taigapy import TaigaClient
+tc = TaigaClient()
+import os
+from genepy import mutations as mut
+from genepy.utils import helper as h
+from genepy import terra
+from genepy.google import gcp
+
+
+def renameColumns(df):
+  """
+  rename some of the main columns names from RSEM, GATK.. to more readable column names
+
+  Args:
+  -----
+    df: the df to rename
+
+  Returns:
+  ------
+    df the renamed df
+  """
+  return df.rename(columns=COLRENAMING)
+
+def loadFromGATKAggregation(refworkspace,  sortby=[SAMPLEID, 'Chromosome', "Start", "End"], 
+                            save_output='', doCleanup=True,
+                            todrop=[], showPlots=False, colname="combined_seg_file",
+                            plotColname="modeled_segments_plot_tumor", tempFolder="temp/",
+                            toremove=["readgroup_ubams", ],
+                            sampleset="all", colRenaming=COLRENAMING):
+  """
+   """
+  wm = dm.WorkspaceManager(refworkspace)
+  if save_output:
+    terra.saveConfigs(refworkspace, os.path.join(save_output, 'terra/'))
+  
+  if doCleanup:
+    print('cleaning up')
+    for val in toremove:
+        wm.disable_hound().delete_entity_attributes('sample', toremove)
+    a = wm.get_samples()
+    e = []
+    for i in a[toremove].values.tolist():
+        if i is not np.nan:
+            e.extend(i)
+    gcp.rmFiles(e)
+  
+  segments = pd.read_csv(wm.get_entities(
+        'sample_set').loc[sampleset, colname], sep='\t').rename(columns=colRenaming)
+ 
+  segments = segments[~segments[SAMPLEID].isin(todrop)].reset_index(drop=True)
+  if "chr" in segments['Chromosome'][0]:
+     segments['Chromosome'] = [i[3:] for i in segments['Chromosome']]
+  segments.Segment_Mean = 2**segments.Segment_Mean
+  segments.Start = segments.Start.astype(int)
+  segments.End = segments.End.astype(int)
+  segments.loc[segments[segments.Chromosome.isin(
+    ['X', 'Y'])].index, 'Segment_Mean'] = segments[segments.Chromosome.isin(['X', 'Y'])]['Segment_Mean']/2
+  segments = segments.sort_values(by=sortby)
+  
+  print("loading "+ str(len(set(segments[SAMPLEID])))+ " rows")
+  if showPlots:
+    # plotting results of CN calls for this new sample set
+    for i, (k, val) in enumerate(wm.get_samples().loc[wm.get_sample_sets().loc[
+        sampleset].samples].iterrows()):
+      plot = val[plotColname]
+      os.system('gsutil cp '+plot+' '+tempFolder)
+      print(k)
+      print(val['arxspan_id'], val['sex'])
+      if i > 30:
+        continue
+      display(Image(os.path.join(tempFolder,plot.split('/')[-1])))
+  return segments
+
+
+def postProcess(refworkspace, sampleset='all', save_output="", doCleanup=True,  sortby=[
```

Comment:
is this the overall postprocess that could be used for CCLF, etc?

Reply:
yes!